mlflow/sklearn/__init__.py

```diff
@@ -932,9 +1390,46 @@ def patched_fit(original, self, *args, **kwargs):
         )
     ]
 
+    def should_patch_class_method(class_def, func_name):
+        try:
+            if hasattr(class_def, func_name):
+                method = getattr(class_def, func_name)
+                return callable(method)
+            return False
+        except:
+            return False
+
+    def patched_IffHasAttrDescriptor_get(self, obj, type=None):
+        from functools import update_wrapper
+        # raise an AttributeError if the attribute is not present on the object
+        if obj is not None:
+            # delegate only on instances, not the classes.
+            # this is to allow access to the docstrings.
+            for delegate_name in self.delegate_names:
+                try:
+                    delegate = sklearn.utils.metaestimators.attrgetter(delegate_name)(obj)
+                except AttributeError:
+                    continue
+                else:
+                    getattr(delegate, self.attribute_name)
+                    break
+            else:
+                sklearn.utils.metaestimators.attrgetter(self.delegate_names[-1])(obj)
+
+        # lambda, but not partial, allows help() to work with update_wrapper
+        if obj is None:  # `cls.method`
+            out = lambda *args, **kwargs: self.fn(*args, **kwargs)
+        else:  # `instance.method`
+            out = lambda *args, **kwargs: self.fn(obj, *args, **kwargs)
+        # update the docstring of the returned function
+        update_wrapper(out, self.fn)
+        return out
```

Comment:
Can we add an inline comment explaining why we're doing this patching?

Reply:
I removed this patch code from this PR., will do that in follow-up PR.